### DISCUSSION TITLE
there is not information about email_change_token_new and  email_change_token_current

### DISCUSSION THREAD

**Discussion by cenullum** [supabase/supabase#27862] [Feature Requests]
# Improve documentation

## Describe the problem

I want to know how to verify both of otps or verify email_change_token_new myself in a rpc function.

![resim](https://github.com/supabase/supabase/assets/57962685/e7fc1ef2-ed85-4d8c-9c95-9e5632ec6e00)
I was trying to update email with current password and otp sent to new email. Actually first I wanted old and new email otp but I couldnt then tried current password instead of old email otp to change email.

I can verify encrypted current password with this code but when I try on email_change_token_new or email_change_token_current I get invalid salt error.

`CREATE OR REPLACE FUNCTION verify_user_password(password text)
RETURNS BOOLEAN SECURITY DEFINER AS
$$
BEGIN
  RETURN EXISTS (
    SELECT id 
    FROM auth.users 
    WHERE id = auth.uid() AND encrypted_password = crypt(password::text, auth.users.encrypted_password)
  );
END;
$$ LANGUAGE plpgsql;`

actually main problem is I couldn't verify_otp both of tokens that sent to new and old emails at same time or one by one
![resim](https://github.com/supabase/supabase/assets/57962685/7d61e416-6cb2-4131-bbab-c5693bb5c1d1)
that is why I try to make a rpc function, I don't know the correct way to verify both of tokens. there isnt information about changing email with otp also with secure email change enabled. therefore I tried without secure email change but also there isnt much information hashed tokens.

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.
